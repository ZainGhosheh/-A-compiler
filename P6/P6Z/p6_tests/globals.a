a:int = 9 + 10;
b:int = 10;
c:bool = eh?;
d:bool = false;