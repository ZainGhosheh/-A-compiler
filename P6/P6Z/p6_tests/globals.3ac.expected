[BEGIN GLOBALS]
a
b
c
d
[END GLOBALS]
[BEGIN <init> LOCALS]
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
[END <init> LOCALS]
fun_<init>: enter <init>
            [tmp0] := 9 ADD64 10
            [a] := [tmp0]
            [b] := 10
            call randBool
            getret [tmp1]
            [c] := [tmp1]
            [d] := 0
lbl_0:      leave <init>
